main.tf

```diff
@@ -175,3 +177,25 @@ resource "gitlab_branch_protection" "default" {
     }
   }
 }
+
+################################################################################
+# Pipeline schedule
+################################################################################
+resource "gitlab_pipeline_schedule" "default" {
+  count  = var.pipeline_schedule.cron != null ? 1 : 0
+  active = var.pipeline_schedule.active
+
+  project        = gitlab_project.default.id
+  description    = var.pipeline_schedule.description
+  ref            = var.pipeline_schedule.ref
+  cron           = var.pipeline_schedule.cron
+  cron_timezone  = var.pipeline_schedule.cron_timezone
+  take_ownership = var.pipeline_schedule.take_ownership
+
+  lifecycle {
```

Comment:
What is the thought behind the preconditon? Would a variable validation not be better?  Moving the validation logic for the cron property to the variable block can make the Terraform configuration cleaner and enforce validation closer to the input source. 

The variable would then result in:

```
variable "pipeline_schedule" {
  type = object({
    active         = optional(bool, true)
    cron           = string
    cron_timezone  = optional(string, "UTC")
    description    = string
    ref            = optional(string, "refs/heads/main")
    take_ownership = optional(bool, false)
  })
  default = null

  validation {
    condition     = can(regex("^([0-5]?[0-9]|\\*) ([0-9]|1[0-9]|2[0-3]|\\*) ([1-9]|[12][0-9]|3[01]|\\*) ([1-9]|1[0-2]|\\*) ([0-6]|\\*)$", var.pipeline_schedule.cron))
    error_message = "The cron expression is not valid. It should be in the format \"0 1 * * *\"."
  }
}
```

Reply:
Oh but thats solvable :D You have to modify the conditional to allow null as well https://github.com/schubergphilis/terraform-aws-mcaf-aurora/blob/master/variables.tf#L55

Then the validation is solved and since its still a non-optional variable the user still has to fill it in in order for Terraform to not fail :-D